### DISCUSSION TITLE
Admin job status: Recognize faces - ALL option

### DISCUSSION THREAD

**Discussion by eygraber** [immich-app/immich#3812] [Q&A] [answered]
In the job-status page on the Recognize Faces card there is a button labeled ALL. Will this destroy any existing face data, or will it run over all of the photos again, using my current face data as part of the model?

**Comment by alextran1502** [comment, marked as answer]
Hello, it will remove existing data, including names and merged faces, then run through the whole dataset again. We don't have the mechanism to train the dataset yet.

**Reply by eygraber** [reply]
So if I tag and merge faces it won't actually train the model to better recognize the face in future photos (yet)?

**Reply by bo0tzz** [reply]
It will improve the recognition for later uploads, but rerunning the job for all faces will do a full reset.

**Reply by eygraber** [reply]
What does the missing option do? Does it rerun the scan on all photos with unlabeled faces?

**Reply by bo0tzz** [reply]
It runs the scan on all assets for which there is no face information at all in the database.

**Reply by eygraber** [reply]
It would be nice if there was a way to run a scan of all unlabeled faces using the info from the currently labeled faces. Is there something tracked for that already?
